### PR TITLE
cormexp fixes

### PR DESCRIPTION
**Pull request by KyleAnthonyShepherd**
General fixing of cormexp behavior.

This is a bit weird. In-game, most (but not all) of the rocket salvos are a burst of 5. Sometimes it is just a burst of 4. 
In the script, it seems like it was intended to be a burst of 4, but due to the default reaim time of 0.5 seconds, the 5th rocket would frequently still be allowed to fire.

So script controlled reload has been removed, and in the unitdef
burst = 5 and burstrate = 0.3 (the old "reload" time)

also set overall reloadtime = 5.9 seconds.
The scripted 5 second reload starts after the 4th rocket.
r1 -> 0.3 seconds -> r2 -> 0.3 seconds -> r3 -> 0.3 seconds -> r4 -> 5 second sleep -> r1

turret turn rate changed to 40,000 for all mex values (instead of variable based on extraction rate)[using turn rate on a +2 mex].

RestoreAfterDelay set to 6 seconds, instead of 10 seconds.

Removed all armor setting states, in favor of just doubling the unit health. Previously, the armor only applied when the unit had it's weapons deployed, and was actually weaker when in the closed up position (in contrast to other units in the game).

Using a different state management scheme, as now extraction and weapon deployment are independent of each other. Along the way, fixed some animation bugs where the columns could get stuck moving at zero speed, and cleaned up the deploy and retract animation (now the weapon pops up, then extends the barrels, instead of doing both motions at the same time)
As a consequence, the cormexp will continue extracting metal even when firing. (If desired, it would be possible to change this behavior back to the old behavior of attacking -> no extraction)

### CODE REVIEW

**Comment by KyleAnthonyShepherd** [comment]
Also, set decoyfor = "cormoho" so opponents still see the cormexp with the "correct" health, and overall make the cormoho better at pretending to be a T2 mex

File: units/CorBuildings/LandDefenceOffence/cormexp.lua [hunk 1]
```diff
@@ -25,7 +26,7 @@ return {
 		icontype = "building",
 		idleautoheal = 5,
 		idletime = 1800,
-		maxdamage = 3900,
+		maxdamage = 7800, --3900
```
Comment (WatchTheFort) [inline units/CorBuildings/LandDefenceOffence/cormexp.lua:29]: We don't need to keep the old values around as comments, if we want to view history, we have source control.

File: scripts/Units/cormexp.bos [hunk 2]
```diff
@@ -5,99 +5,51 @@
 
 piece base, column, top, turret1, turret2, barrel, flare1, flare2, flare3, aimpoint;
 
-static-var  ExtractionSpeed, ArmorUp, restore_delay,
-            gun_1, gun_2, Static_Var_8, bAiming, statechg_DesiredState, statechg_StateChanging;
+static-var  ExtractionSpeed, restore_delay,
+            gun_1, gun_2, statechg_DesiredState, statechg_StateChanging, column_going_up, weapon_state;
```
Comment (WatchTheFort) [inline scripts/Units/cormexp.bos:9]: `statechg_DesiredState` and `statechg_StateChanging` are unused and can be removed

File: units/CorBuildings/LandDefenceOffence/cormexp.lua (hunk 1 above)
Comment (KyleAnthonyShepherd) [reply units/CorBuildings/LandDefenceOffence/cormexp.lua]: Thanks for catching that

File: scripts/Units/cormexp.bos (hunk 2 above)
Comment (KyleAnthonyShepherd) [reply scripts/Units/cormexp.bos]: addressed

**Comment by badosu** [comment]
Thank you!

**Pull request merged**